FirebaseInstallations/Source/Library/FIRInstallationsItem.m

```diff
@@ -72,22 +72,22 @@ + (NSString *)generateFID {
 
   NSData *uuidData = [NSData dataWithBytes:uuidBytes length:16];
 
-  uint8_t prefix = 0b01110000;
-  NSMutableData *fidData = [NSMutableData dataWithBytes:&prefix length:1];
+  uint8_t uuidLast4Bits = uuidBytes[15] & 0b00001111;
```

Comment:
UUID_LENGTH-1

Reply:
I defined a local var `NSUInteger UUIDLength = sizeof(uuid_t);`, so the UUIDBytes[UUIDLength - 1] would not crash even if actual `UUID` length changes.